### ISSUE TITLE
excuse me, two questions want to consult

### ISSUE THREAD

**Issue by perfectshils** [ZainNasrullah/music-artist-classification-crnn#2]
thank you for your code and paper, they give me some new ideas for artist classification. But I still have two questions about it:
1, It seems that I can't find the code about switch the method to select song level or frame level.
2, I ran this program, but got the finally F1 score with only about 0.7 when using  album split with 3s rather than above 0.9 in your paper.

so, I wonder that,  did I miss some details when runing this program?

**Comment by ZainNasrullah** [comment]
Hi, if I recall correctly:

1. The code should be generating metrics for both song and frame level. In [main.py](https://github.com/ZainNasrullah/music-artist-classification-crnn/blob/7e17d201ed2c4dd11b5fd6587c7534b4f1d494ed/main.py#L47), the trainer returns both scores (frame-level) and pooling_scores (song-level). Apologies for the disconnect in terminology, we clarified some items for the paper but didn't have a chance to reflect this in the code. The respective output csvs '{}/{}_score.csv' and '{}/{}_pooled_score.csv' will contain the stuff you are looking for.

2. The scores > 0.9 are only achieved at the song-level with the song-split (first row of table 5 in the paper). When using the album split, the performance is notably worse--likely--due to overfitting on production details (i.e., the producer effect) or model capacity. This is still an open problem worth investigating :)

**Comment by perfectshils** [comment]
Thank you for your patient answser. I‘m so sorry, I described the second question wrong, it is song split rather than album split. here I uplode the screenshots about the program result. The results are actually not >0.9

![1575271796(1)](https://user-images.githubusercontent.com/17350975/69938845-5012da80-1519-11ea-9d34-e61697614400.png)
![1575271808(1)](https://user-images.githubusercontent.com/17350975/69938847-50ab7100-1519-11ea-95a7-733489e78add.png)

**Comment by perfectshils** [comment]
![1575271796(1)](https://user-images.githubusercontent.com/17350975/69939071-e34c1000-1519-11ea-9571-f3b341b354d9.png)

**Comment by perfectshils** [comment]
For this problem, I don't know what went wrong, so need your help and advice

**Comment by ZainNasrullah** [comment]
It's been a couple of months since I've looked at this, but at the time of testing we were consistently seeing scores > 0.9.

A couple of thoughts:
1. Your results look a lot like the album split rather than the song split. Can you confirm that the album_split flag is set to false in main.py? Can you also share your main.py? 

2. It's possible that there is some discrepancy elsewhere (silent failure) such as a version mismatch (do any of your versions for librosa, keras or tensorflow differ from the requirements?) or a bug in the code regarding the splitting. If you look at the song names, are the train, test and validation splits as expected?

3. Are you training the model until early stopping occurs or for a fixed number of iterations? It's possibly you are getting unlucky initialization but I don't recall seeing those results.

**Comment by perfectshils** [comment]
Yes , it is song splite, in main.py ,the parameter album_split=False,  I am planing to run it again tonight, and change the early stop from 10 to 50, maybe the 'earlystop' is too early.... I'll see if there are any changes in the results tomorrow......

**Comment by ZainNasrullah** [comment]
Hi, I re-ran some of these tests for the 3s / 94 frame case and am getting results consistent with the paper (see attached for a few samples). I did not modify any settings in main.py other than:
- slice_lengths = [94]
- random_state_list = [0, 11, 21, 42]
- iterations = 4

Consequently, I believe early stopping is not the issue. Assuming main.py has not been modified and the data is generated as outlined in the usage section, this leads me to believe that there is perhaps a version or OS mismatch (I am using ubuntu 18.04 in my tests) with respect to one of the dependencies on your machine. 

[20_94_42.txt](https://github.com/ZainNasrullah/music-artist-classification-crnn/files/3913592/20_94_42.txt)
[20_94_21.txt](https://github.com/ZainNasrullah/music-artist-classification-crnn/files/3913593/20_94_21.txt)
[20_94_11.txt](https://github.com/ZainNasrullah/music-artist-classification-crnn/files/3913594/20_94_11.txt)
[20_94_0.txt](https://github.com/ZainNasrullah/music-artist-classification-crnn/files/3913595/20_94_0.txt)

**Comment by perfectshils** [comment]
Maybe it's a version or OS mismatch. Not only did I have problems with 3s mentioned before, I also tried 1s, 5s and 10s, and the final result was all about  0.15-0.2 lower than that mentioned in the paper. I'll try to debug the parameters again, and if the results aren't good enough, I'll give up...

All in all,  thank you very much !

[20_32_0.txt](https://github.com/ZainNasrullah/music-artist-classification-crnn/files/3914362/20_32_0.txt)
[20_91_0.txt](https://github.com/ZainNasrullah/music-artist-classification-crnn/files/3914363/20_91_0.txt)
[20_94_0.txt](https://github.com/ZainNasrullah/music-artist-classification-crnn/files/3914364/20_94_0.txt)
[20_157_0.txt](https://github.com/ZainNasrullah/music-artist-classification-crnn/files/3914365/20_157_0.txt)

**Comment by ZainNasrullah** [comment]
You're welcome @perfectshils, and I apologize for not being able to provide further assistance. Since I am not able to reproduce, the problem could be anywhere between data quality and the keras/tf version. If you end up figuring it out, please do report back here as others may find your insights helpful.
